src/v/cloud_storage/remote_partition.cc

```diff
@@ -1068,7 +1070,8 @@ remote_partition::aborted_transactions(offset_range offsets) {
             // up front at the start of the function.
             auto segment_unit = co_await materialized().get_segment_units(
               std::nullopt);
-            auto path = stm_manifest.generate_segment_path(*it);
+            auto path = stm_manifest.generate_segment_path(
+              *it, _manifest_view->path_provider());
```

Comment:
the path provider from `_manifest_view` is used with `stm_manifest` which has its own `path_provider` (I know that there is one path provider per partition but the code is written in such way as if everything could have a different instance of `path_provider` but on the other hand it doesn't respect this in all cases, like here)

Reply:
We discussed a bit and agreed that we could plumb a path provider directly into remote_partition.

I'll at least defend this current impl a bit, but I'm happy to make that change if you still think it's less confusing.

The async manifest view is the root of the read path, in the same way that ntp archiver is the root of the write path. Ultimately they’re both rooted in the STM’s path provider, but at least there is a conceptual hierarchy , not just "dumb plumbing"